opencsp/app/sofast/test/test_ProcessSofastFixed.py

```diff
@@ -56,37 +110,78 @@ def setUpClass(cls):
-        cls.process_sofast_fixed.load_measurement_data(measurement)
+        self.sofast_single_facet = ProcessSofastFixed(orientation, camera, dot_locs)
+        self.sofast_single_facet.params = params
+        self.sofast_single_facet.load_measurement_data(measurement)
 
         # Process
-        cls.process_sofast_fixed.process_single_facet_optic(surface)
+        pt_known = measurement.origin
+        xy_known = (0, 0)
+        self.sofast_single_facet.process_single_facet_optic(facet_data, surface, pt_known, xy_known)
+
+    def test_save_facet_ensemble_as_hdf(self):
+        """Tests saving to HDF file"""
+        self.sofast_facet_ensemble.save_to_hdf(join(self.save_dir, 'data_calculation_facet_ensemble.h5'))
 
-    def test_save_as_hdf(self):
+    def test_save_single_facet_as_hdf(self):
         """Tests saving to HDF file"""
-        self.process_sofast_fixed.save_to_hdf(join(self.save_dir, 'data_calculation_sofast_fixed.h5'))
+        self.sofast_single_facet.save_to_hdf(join(self.save_dir, 'data_calculation_single_facet.h5'))
 
-    def test_save_figure(self):
-        """Tests generating and saving a figure"""
+    def test_save_facet_ensemble_slope_figure(self):
+        """Tests genreating and saving a figure of a facet ensemble"""
         figure_control = rcfg.RenderControlFigure(tile_array=(1, 1), tile_square=True)
         axis_control_m = rca.meters()
 
         fig_mng = fm.setup_figure(figure_control, axis_control_m, title='')
-        mirror = self.process_sofast_fixed.get_mirror('bilinear')
-        mirror.plot_orthorectified_slope(res=0.002, clim=7, axis=fig_mng.axis)
-        fig_mng.save(self.save_dir, 'sofast_fixed_orthorectified_slope_magnitude', 'png')
+        ensemble = self.sofast_facet_ensemble.get_optic('bilinear')
+        ensemble.plot_orthorectified_slope(res=0.002, clim=7, axis=fig_mng.axis)
+        fig_mng.save(self.save_dir, 'slope_magnitude_facet_ensemble', 'png')
 
-    def test_slopes_xy(self):
+    def test_save_single_facet_slope_figure(self):
+        """Tests generating and saving a figure of a single facet"""
+        figure_control = rcfg.RenderControlFigure(tile_array=(1, 1), tile_square=True)
+        axis_control_m = rca.meters()
+
+        fig_mng = fm.setup_figure(figure_control, axis_control_m, title='')
+        facet = self.sofast_single_facet.get_optic('bilinear')
+        facet.plot_orthorectified_slope(res=0.002, clim=7, axis=fig_mng.axis)
+        fig_mng.save(self.save_dir, 'slope_magnitude_single_facet', 'png')
+
+    def test_slopes_xy_facet_ensemble(self):
+        """Tests slope data"""
+        for idx_facet in range(self.sofast_facet_ensemble.num_facets):
+            with self.subTest(i=idx_facet):
+                np.testing.assert_allclose(
+                    self.sofast_facet_ensemble.data_calculation_facet[idx_facet].slopes_facet_xy,
+                    self.exp_slopes_xy_facet_ensemble[idx_facet],
+                    rtol=0,
+                    atol=1e-6,
```

Comment:
brief hashtag comment explaining tolerances would be helpful for maintainers

Reply:
Done